test/_files/expected-codegen-results/factories/with-params.php

```diff
@@ -28,10 +31,15 @@ public function create(ContainerInterface $container, array $options = [])
                 array_key_exists('anyDep', $options) ? $options['anyDep'] : null,
             ];
 
+        /** @psalm-suppress MixedArgument */
         return new \LaminasTest\Di\TestAsset\Constructor\MixedArguments(...$args);
```

Comment:
The interface type would correctly be: `public function create(ContainerInterface $container, PartialArray<ConstructArgsArray<T>> $options = []): T`

This can only be achieved with a psalm plug-in as far as I know. But this is out of scope of this PR.
To avoid additional overhead in these generated classes, a suppress is added and responsibility is passed to PHP's type system until a better solution can be provided.